akd/src/directory.rs

```diff
@@ -479,20 +479,14 @@ where
             return Err(AkdError::Storage(StorageError::NotFound(msg)));
         }
 
-        let mut update_proofs = Vec::<UpdateProof>::new();
-        let mut start_version = user_data[0].version;
-        let mut end_version = 0;
-        for user_state in &user_data {
-            // Ignore states in storage that are ahead of current directory epoch
-            if user_state.epoch <= current_epoch {
-                let proof = self
-                    .create_single_update_proof(akd_label, user_state)
-                    .await?;
-                update_proofs.push(proof);
-                start_version = std::cmp::min(user_state.version, start_version);
-                end_version = std::cmp::max(user_state.version, end_version);
-            }
-        }
+        let versions = &user_data
+            .iter()
+            .filter(|user_state| user_state.epoch <= current_epoch)
+            .map(|user_state| user_state.version)
+            .collect::<Vec<u64>>();
+        let first_version = user_data[0].version;
+        let start_version = versions.iter().min().copied().unwrap_or(first_version);
```

Comment:
NOTE: this is slightly different than the original code as previously, it would find the min of `[first_version] ++ min_state_version`, whereas now it just uses `first_version` if no versions were found in `min_state_version`. Maybe will be best to match to old behavior exactly

Reply:
Hmm yeah I would say let's just match the old behavior exactly. The min / max computation won't be expensive, and  rather than using the .iter().filter() syntax I think it's fine to just have the explicit loop as before.

Reply:
updated to chaining the elements to a list so now the logic should be in parity. I can switch back to an explicit loop if preferred.

Reply:
Yeah let's just do the explicit loop still... the presence of unwrap()s makes me uneasy anyway.